### PR TITLE
Update compiledLambdaMap in RowExpressionCompiler for SQL functions

### PR DESCRIPTION
**Pull request by rongrong**
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/gen/LambdaBytecodeGenerator.java [hunk 1]
```diff
@@ -101,6 +102,19 @@ private LambdaBytecodeGenerator()
...
 
+    public static Map<LambdaDefinitionExpression, CompiledLambda> generateMethodsForLambda(
+            ClassDefinition containerClassDefinition,
+            CallSiteBinder callSiteBinder,
+            CachedInstanceBinder cachedInstanceBinder,
+            RowExpression expression,
+            Metadata metadata,
+            SqlFunctionProperties sqlFunctionProperties,
+            String methodNamePrefix,
+            Set<LambdaDefinitionExpression> compiledLambdas)
```
Comment (wenleix) [inline presto-main/src/main/java/com/facebook/presto/sql/gen/LambdaBytecodeGenerator.java:113]: nit: what about something like `alreadyCompiledLambdas` ?

**Comment by caithagoras** [comment]
I don't have context about this PR - are there anything failing without this PR - it's unclear to me with the commit message?

File: presto-main/src/main/java/com/facebook/presto/sql/gen/LambdaBytecodeGenerator.java (hunk 1 above)
Comment (rongrong) [reply presto-main/src/main/java/com/facebook/presto/sql/gen/LambdaBytecodeGenerator.java:113]: Renamed to `existingCompiledLambdas`. "Compiled" kinda implied "already" so it's a little weird.

**Comment by rongrong** [comment]
> I don't have context about this PR - are there anything failing without this PR? - It's unclear to me with the commit message.

With CSE, we could have multiple projections within the same generated bytecode class. All lambdas in these projections will be precompiled as methods in this class. If there are common lambdas, the existing code will try to generate them separately, which results in name collisions. To avoid this, we change the `compiledLambdaMap` to be mutable and only compile lambdas that's not generated yet, and keep updating the same map.

**Pull request merged**